e3sm_to_cmip/_logger.py

```diff
@@ -1,68 +1,75 @@
+import inspect
 import logging
 import os
-import time
-from datetime import datetime
+from datetime import datetime, timezone
 
-from pytz import UTC
 
+DEFAULT_LOG_LEVEL = logging.DEBUG
+DEFAULT_LOG_DIR = "e2c_logs"
+DEFAULT_LOG = f"{DEFAULT_LOG_DIR}/e2c_root_log-{datetime.now(timezone.utc).strftime('%Y%m%d_%H%M%S_%f')}.log"
 
-def _setup_root_logger() -> str:  # pragma: no cover
-    """Sets up the root logger.
+def _logger(name=None, logfilename=DEFAULT_LOG, set_log_level=None, to_console=False, to_logfile=False, propagate=False):
```

Comment:
```suggestion
def _logger(name=None, logfilename=DEFAULT_LOG, set_log_level=None, to_console=False, to_logfile=False, propagate=False):
```

Suggestions
1. Format function definition using `black` and use type annotations.
2. Rename `set_log_level` to `log_level` and pass `logging` levels (e.g., `logging.level`).
3. Update docstring based on previous changes

Reply:
Weird - the "suggested change" (above) looks like no change, due to wrapping."

Addressed 2 and 3, and performed black.

Are type annotations really needed?  There are scored of function definitions in mpas.py, etc, and none of them employ type annotations.